frmts/vrt/pixelfunctions.cpp

```diff
@@ -1165,6 +1212,99 @@ CPLErr InterpolatePixelFunc( void **papoSources, int nSources, void *pData,
     return CE_None;
 }
 
+static const char pszReplaceNoDataPixelFuncMetadata[] =
+"<PixelFunctionArgumentsList>"
+"   <Argument type='builtin' value='NoData' />"
+"   <Argument name='to' type='double' description='New NoData value to be replaced' default='nan' />"
+"</PixelFunctionArgumentsList>";
+
+static CPLErr ReplaceNoDataPixelFunc( void **papoSources, int nSources, void *pData,
+                               int nXSize, int nYSize,
+                               GDALDataType eSrcType, GDALDataType eBufType,
+                               int nPixelSpace, int nLineSpace, CSLConstList papszArgs ) {
+    /* ---- Init ---- */
+    if( nSources != 1 ) return CE_Failure;
+    if (GDALDataTypeIsComplex(eSrcType))
+    {
+        CPLError(
+          CE_Failure, CPLE_AppDefined, "replace_nodata cannot convert complex data types");
+        return CE_Failure;
+    }
+
+    double dfOldNoData, dfNewNoData = NAN;
+    if ( FetchDoubleArg(papszArgs, "NoData", &dfOldNoData) != CE_None ) return CE_Failure;
+    if ( FetchDoubleArg(papszArgs, "to", &dfNewNoData, &dfNewNoData) != CE_None ) return CE_Failure;
+
+    if (!GDALDataTypeIsFloating(eBufType) && std::isnan(dfNewNoData))
+    {
+        CPLError(
+          CE_Failure, CPLE_AppDefined, "Using nan requires a floating point type output buffer");
+        return CE_Failure;
+    }
+
+    /* ---- Set pixels ---- */
+    size_t ii = 0;
+    for( int iLine = 0; iLine < nYSize; ++iLine )
+    {
+        for( int iCol = 0; iCol < nXSize; ++iCol, ++ii )
+        {
+            double dfPixVal = GetSrcVal(papoSources[0], eSrcType, ii);
+            if (dfPixVal == dfOldNoData || std::isnan(dfPixVal)) dfPixVal = dfNewNoData;
+
+            GDALCopyWords(
+                    &dfPixVal, GDT_Float64, 0,
+                    static_cast<GByte *>(pData) + static_cast<GSpacing>(nLineSpace) * iLine +
+                    iCol * nPixelSpace, eBufType, nPixelSpace, 1);
+        }
+    }
+
+    /* ---- Return success ---- */
+    return CE_None;
+}
+
+static const char pszScalePixelFuncMetadata[] =
+"<PixelFunctionArgumentsList>"
+"   <Argument type='builtin' value='offset' />"
+"   <Argument type='builtin' value='scale' />"
+"</PixelFunctionArgumentsList>";
+
+static CPLErr ScalePixelFunc( void **papoSources, int nSources, void *pData,
+                               int nXSize, int nYSize,
+                               GDALDataType eSrcType, GDALDataType eBufType,
+                               int nPixelSpace, int nLineSpace, CSLConstList papszArgs ) {
+    /* ---- Init ---- */
+    if( nSources != 1 ) return CE_Failure;
+    if (GDALDataTypeIsComplex(eSrcType))
```

Comment:
In principle scale and offset could also be applied to a complex band, even if we have the limitation that ``offset`` an d``scale`` factors con be only real numbers.

Reply:
I will push this in its current form - I have never used complex numbers Datasets and I can't evaluate if these make sense or not